### PR TITLE
231473 image render fix

### PR DESCRIPTION
**Pull request by adunndevster**
I consolidated the ImageExtension code @yufeih !

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6044)

### CODE REVIEW

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ImageExtension.cs [hunk 1]
```diff
@@ -94,23 +94,30 @@ public bool TryProcessAttributes(IDictionary<string, string> attributes, out Htm
             {
                 htmlAttributes.AddProperty("alt", alt);
             }
-            var id = GetHtmlId(block);
+            var id = GetHtmlId(markdownObject);
             if(type == "complex") htmlAttributes.AddProperty("aria-describedby", id);
 
             RenderDelegate = (renderer, obj) =>
             {
+                dynamic blockOrInline;
```
Comment (yufeih) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ImageExtension.cs:102]: Is there a way to not using `dynamic` here? `dynamic` is likely to hurt the performance of this method: https://stackoverflow.com/questions/7478387/how-does-having-a-dynamic-variable-affect-performance

Comment (adunndevster) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ImageExtension.cs]: Does it look better now @yufeih ?

Comment (yufeih) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ImageExtension.cs]: Yep, much better. There are two tests failing though.

Comment (adunndevster) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ImageExtension.cs]: @yufeih - It looks like those tests are passing now. I noticed the regression diff tests (is that what you call them?) were not passing so I downloaded one of the diff files, and it had lots of this:
image

That is now what the image output should be, so we should be good to merge this, yes?

Comment (yufeih) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ImageExtension.cs]: yes

**Pull request merged**